mobipick_gazebo/nodes/robotiq_2f_140_command_bridge.py

```diff
@@ -56,47 +56,73 @@ def set_jt_goal(self, position, effort=30.0):
     def execute_cb(self, goal):
         # helper variables
         rospy.loginfo(
-            "Recieved gripper command with position=%.3f and effort=%.3f"
+            "Received gripper command with position=%.3f and effort=%.3f"
             % (goal.command.position, goal.command.max_effort)
         )
         r = rospy.Rate(10)
         success = True
         self.set_jt_goal(-0.76 / 0.14 * goal.command.position + 0.76, goal.command.max_effort)
         self._jt_client.send_goal(self._actual_goal)
 
-        # start executing the action
-        while not self._jt_client.get_state() == actionlib.SimpleGoalState.ACTIVE:
-            # check that preempt has not been requested by the client
+        # Start executing the action
+        while not rospy.is_shutdown():
+            # Check if the action client has finished
+            state = self._jt_client.get_state()
+            if state in [
+                actionlib.GoalStatus.SUCCEEDED,
+                actionlib.GoalStatus.ABORTED,
+                actionlib.GoalStatus.PREEMPTED,
+                actionlib.GoalStatus.REJECTED,
```

Comment:
This list is missing one possible terminal state: `RECALLED`. I've never seen this in the wild, and I'm not even sure it's possible with the SimpleActionServer, but it should be added to the list.